libs/parallelism/futures/include/hpx/futures/futures_factory.hpp

```diff
@@ -106,9 +108,7 @@ namespace hpx { namespace lcos { namespace local {
             {
                 this->check_started();
 
-                typedef typename Base::future_base_type future_base_type;
-                future_base_type this_(this);
-
+                hpx::intrusive_ptr<base_type> this_(this);
```

Comment:
These I don't know about. They may indeed be needed.

Reply:
Yes, this was simply a bug as it did store the plain pointer instead of the `intrivise_ptr`.